src/Microsoft.Azure.SignalRService/ServiceHubContextBackend.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.SignalRService
+{
+    public enum ServiceHubContextBackend
```

Comment:
>ServiceHubContextBackend [](start = 16, length = 24)

is this some `TransportType`?

Reply:
No, but it is similar to `HttpTransportType`.
HttpTransportType.Longpolling ~= ServiceHubContextBackend.RestV1 
HttpTransportType.WebSockets == ServiceHubContextBackend.Websocket

How about I extend HttpTransportTypeto ServiceTransportType which contains additional field `RestV1`

Reply:
better then current naming

---
In reply to: [242399599](https://github.com/Azure/azure-signalr/pull/289#discussion_r242399599) [](ancestors = 242399599)

Reply:
but I don't think `RestV1` sounds like a transport type..

---
In reply to: [242412014](https://github.com/Azure/azure-signalr/pull/289#discussion_r242412014) [](ancestors = 242412014,242399599)

Reply:
How about changing `RestV1` to `Transient`. Since compared to `Websocket` mode, sending messages is transient for each request in `Rest` mode.

Since the version is removed from the name, we should add an additional enum for rest api version. 


``` C#
var serviceManager = new ServiceManagerBuilder()
                    .WithCredential(connectionString) 
                    .WithOptions(option => 
                    {
                        option.ServiceTransportType = ServiceTransportType.Transient,
                        option.ServiceTransportVersion = ServiceTransportVersion.V1 // optional, default is v1
                    })
                    .Build();
```